dattri/func/ihvp.py

```diff
@@ -1225,11 +1254,13 @@ def cached_forward(self: torch.nn.Module, *args, **kwrds) -> torch.Tensor:
         cache.check_type()
         cache.retain_grad()
         return outputs
+
     return cached_forward
 
 
 class MLPCache:
     """Cache of input and output variables in a MLP layer."""
```

Comment:
I am thinking whether we can just use forward_hook for this part.
I know this will probably not be addressed by this PR, but just for simplicity.

Reply:
Yes that's our plan, will do for next PR. Had some discussion with @sx-liu already.